src/sentry/middleware/auth.py

```diff
@@ -48,6 +49,11 @@ def process_request(self, request):
         else:
             request.user = SimpleLazyObject(lambda: get_user(request))
 
+        # Update last_login of user if it exceeds the defined period
+        if can_update_last_login(request.user):
+            from sentry.signals import keep_user_logged_in
```

Comment:
If we move this import to the module level, does it trigger a recursive import? Trying to understand why it's being imported inside this `if` otherwise.